logging/nxscope/nxscope_pser.c

```diff
@@ -0,0 +1,297 @@
+ * +-----+----+----+----+------------+------+------+
+ * | 0   | 1  | 2  | 3  | ...        | n+1  | n+2  |
+ * +-----+----+----+----+------------+------+------+
+ *
+ * [1] - always little-endian
+ * [2] - always big-endian
+ */
+
+/* Nxscope header */
+
+begin_packed_struct struct nxscope_hdr_s
+{
+  uint8_t  sof;                 /* SOF */
+  uint16_t len;                 /* Frame length */
+  uint8_t  id;                  /* Frame ID */
+} end_packed_struct;
+
+/* Nxscope footer */
+
+begin_packed_struct struct nxscope_footer_s
+{
+  uint16_t crc16;               /* check sum (see nxscope_frame_final()) */
+} end_packed_struct;
+
+/****************************************************************************
+ * Private Function Protototypes
+ ****************************************************************************/
+
+static int nxscope_frame_get(FAR struct nxscope_proto_s *p,
+                             FAR uint8_t *buff, size_t len,
+                             FAR struct nxscope_frame_s *frame);
+static int nxscope_frame_final(FAR struct nxscope_proto_s *p,
+                               uint8_t id,
+                               FAR uint8_t *buff, FAR size_t *len);
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+struct nxscope_proto_ops_s g_nxscope_proto_ser_ops =
+{
+  .frame_get     = nxscope_frame_get,
+  .frame_final   = nxscope_frame_final,
+};
+
+struct nxscope_proto_s g_nxscope_proto_ser =
+{
+  .priv          = NULL,
+  .ops           = &g_nxscope_proto_ser_ops,
+  .footlen       = NXSCOPE_CRC_LEN,
+  .hdrlen        = NXSCOPE_HDR_LEN
+};
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: nxscope_hdr_fill
+ ****************************************************************************/
+
+static void nxscope_hdr_fill(FAR uint8_t *buff, uint8_t id, uint16_t len)
+{
+  FAR struct nxscope_hdr_s *hdr  = NULL;
+
+  hdr = (FAR struct nxscope_hdr_s *) buff;
+
+  hdr->sof = NXSCOPE_HDR_SOF;
+
+  /* Length always as little endian */
+
+#ifdef CONFIG_ENDIAN_BIG
+  hdr->len = bswap_16(len);
+#else
+  hdr->len = len;
+#endif
+
+  hdr->id  = id;
+}
+
+/****************************************************************************
+ * Name: nxscope_frame_get
+ ****************************************************************************/
+
+static int nxscope_frame_get(FAR struct nxscope_proto_s *p,
+                             FAR uint8_t *buff, size_t len,
+                             FAR struct nxscope_frame_s *frame)
+{
+  FAR struct nxscope_hdr_s *hdr = NULL;
+  uint16_t                  crc = 0;
+  int                       ret = OK;
+  int                       i   = 0;
```

Comment:
```suggestion
  FAR struct nxscope_hdr_s *hdr;
  uint16_t                  crc;
  int                       ret = OK;
  int                       i;
```

Reply:
I just don't like uninitialized variables, so I would like leave it here (and in some other places)